tests/contrib/test_pydantic.py

```diff
@@ -0,0 +1,246 @@
+import dataclasses
+import uuid
+from datetime import datetime, timedelta
+from ipaddress import IPv4Address
+from typing import Annotated, Any, List, Sequence, Tuple, TypeVar
+
+from annotated_types import Len
+from pydantic import BaseModel, Field, WithJsonSchema
+
+from temporalio import activity, workflow
+from temporalio.client import Client
+from temporalio.contrib.pydantic import pydantic_data_converter
+from temporalio.worker import Worker
+
+SequenceType = TypeVar("SequenceType", bound=Sequence[Any])
+ShortSequence = Annotated[SequenceType, Len(max_length=2)]
+
+
+class MyPydanticModel(BaseModel):
+    ip_field: IPv4Address
+    datetime_field: datetime
```

Comment:
Can we add `datetime.date`, `datetime.time`, and `datetime.timedelta` in here? And maybe `uuid.UUID` while we're at it since we have restrictions in that module too.

Reply:
Done, the test suite has been significantly expanded.